src/serde/de.rs

```diff
@@ -18,26 +21,28 @@ where
     T: DeserializeOwned,
     I: IonInput,
 {
+    let mut ion_encoding = IonEncoding::default();
     let mut reader = SystemReader::new(AnyEncoding, input);
-    let item = reader.next_item()?;
-    match item {
-        SystemStreamItem::VersionMarker(marker) => {
-            // Note that this uses the Ion version with which the IVM was encoded rather than
-            // the Ion version the stream is switching to. We can do this because the format
-            // (i.e text or binary) stays the same when the version changes.
-            // TODO: Use new encoding, once we have APIs to get new/old encodings for the marker.
-            let is_human_readable = marker.encoding().is_text();
-            let value = reader.expect_next_value()?;
-            let value_deserializer = ValueDeserializer::new(&value, is_human_readable);
-            T::deserialize(value_deserializer)
-        }
-        SystemStreamItem::Value(value) => {
-            let value_deserializer = ValueDeserializer::new(&value, true);
-            T::deserialize(value_deserializer)
```

Comment:
🗺️ This old implementation would return an error if an LST preceded the first value.